src/Workspaces/Core/Portable/Utilities/EditDistance.cs

```diff
@@ -108,8 +108,9 @@ public int GetEditDistance(string target, int threshold = int.MaxValue)
         private static readonly ThreadLocal<int[,]> t_matrixPool = 
             new ThreadLocal<int[,]>(() => InitializeMatrix(new int[MaxMatrixPoolDimension, MaxMatrixPoolDimension]));
 
-        private static ThreadLocal<Dictionary<char, int>> t_dictionaryPool = 
-            new ThreadLocal<Dictionary<char, int>>(() => new Dictionary<char, int>());
+        private const int LastSeenIndexLength = 128;
+        private static ThreadLocal<int[]> t_lastSeenIndexPool = 
```

Comment:
interesting.  I both love it... but hate using unsafe... i'll postpone this for now.  But def an interesting idea.